src/tasks/submit/task.rs

```diff
@@ -151,27 +157,26 @@ impl SubmitTask {
                 "SubmitTask::retrying_send",
                 retries = bumpable.bump_count(),
                 nonce = bumpable.req().nonce,
+                expected_slot,
+                start = window.start,
+                end = window.end,
+                unix_now = utils::now(),
             );
 
-            let inbound_result = match self.send_transaction(req).instrument(span.clone()).await {
-                Ok(control_flow) => control_flow,
-                Err(error) => {
-                    if let Some(value) = self.slot_still_valid(expected_slot) {
-                        return value;
-                    }
-                    // Log error and retry
-                    error!(%error, "error handling inbound block");
-                    ControlFlow::Retry
-                }
-            };
+            // Check at the top of the loop if the slot is still valid. This
+            // will prevent unnecessary retries if the slot has changed.
+            check_slot_still_valid!(self, expected_slot);
 
-            let guard = span.entered();
+            let inbound_result = self
+                .send_transaction(req)
+                .instrument(span.clone())
+                .await
+                .inspect_err(|e| error!(error = %e, "sending transaction"))
```

Comment:
this was probably written this way because it's implied with the error i assume, but i still think it can be confusing at first glance
```suggestion
                .inspect_err(|e| error!(error = %e, "error sending transaction"))
```